### ISSUE TITLE
Error withdrawing offer

### ISSUE THREAD

**Issue by sparrowDom** [OriginProtocol/origin#3305]
On mainnet when withdrawing offer Id: 1-000-3126-4 I got a failed transaction with this error: 
**Fix for this issue:** https://github.com/OriginProtocol/origin/pull/3725
![Screenshot_20190908-233006](https://user-images.githubusercontent.com/579910/64494761-33a43080-d291-11e9-8dab-434cd98fb701.jpg)

**Comment by sparrowDom** [comment]
Not sure if this error is related: https://github.com/OriginProtocol/origin/issues/3283

**Comment by mikeshultz** [comment]
I don't think it's related.  Curious error.  

CC @tomlinton

**Comment by tomlinton** [comment]
This'll happen if you do a web3.eth.sendTransaction from an account that is not "unlocked" in web3, i.e. web3.eth.accounts.wallet.add or web3.eth.accounts.privateKeyToAccount has not been called with an account with the some from address.

Not sure what/why was trying to do that though, maybe in _txHelper.js.

**Comment by mikeshultz** [comment]
I tested editing a listing on catcher 0.22 on Rinkeby.  Worked okay for me.  Mine went over the relayer.

**Comment by micahalcorn** [comment]
Not likely specific to withdrawing 🐘

**Comment by franckc** [comment]
I just tested editing a listing, which was consistently failing for me before, and it worked perfectly for me. I'm ok closing this issue if @micahalcorn can't reproduce it.

**Comment by micahalcorn** [comment]
Sure 🤞

**Comment by sparrowDom** [comment]
Franck could you see if you can reproduce the error using the default mnemonic: 
`candy maple cake sugar pudding cream honey rich smooth crumble sweet treat`

then going to staging dapp: 
- and navigating to active listings
- and trying to [edit the listing with id 4-000-127](https://dapp.staging.originprotocol.com/#/listing/4-000-127/edit/details)

**Comment by sparrowDom** [comment]
I think I have gotten closer to the root of the problem. First of all this can only be reproduced with mobile device and rinkeby config combination - local config on mobile and rinkeby on desktop work fine.

Editing the listing above causes the `tx_helper` to send the transaction without proxy enabled with justification [no proxy specified](https://github.com/OriginProtocol/origin/blob/master/packages/graphql/src/mutations/_txHelper.js#L133) This code path doesn't execute if you create a listing on staging and edit it, since the proxy option is on.

Setting local dapp to Rinkeby config (http://localhost:3000/rinkeby#/) and altering hardcoded [proxyAccountsEnabled](https://github.com/OriginProtocol/origin/blob/master/packages/graphql/src/configs/rinkeby.js#L62) to false starts to cause the original error also on listing creation - besides listing edit. 

Interestingly the same settings in case of identity publishing allow a user to execute the transaction and even [process transaction](https://github.com/OriginProtocol/origin/blob/master/packages/mobile-bridge/src/index.js#L16) gets called. But the changes to the identity are not visible in profile.

For listing creation and listing edit our [Zero client provider](https://github.com/OriginProtocol/origin/blob/master/packages/mobile-bridge/src/index.js#L100) with alchemy Rinkeby rpcUrl configured doesn't trigger the signing of transaction. I think it is trying to call `sendTransaction` on alchemy provider without first calling `signTransaction` locally and then sending it to Alchemy via `sendRawTransaction`. 

So to sum up, seems that mobile client with alchemy as the main provider and proxy off is causing problems.

@mikeshultz and @tomlinton any ideas what is going on here?

**Comment by mikeshultz** [comment]
Adding for reference, on an `eth_sendTransaction` RPC call, hooked wallet should call `processTransaction`.

https://github.com/MetaMask/web3-provider-engine/blob/e28d4417bf0a6bfc4a0aea9035d71ddc6dc2ae2e/subproviders/hooked-wallet.js#L113-L119

**Comment by sparrowDom** [comment]
[This pull request](https://github.com/OriginProtocol/origin/pull/3725/files) fixes the transaction not hitting the `processTransaction` on our mobile provider. The result is that the:
- transaction signature confirmation UI pops up on the mobile device
- the transaction hash is received by the txHelper. But the transaction is not recorded on the chain. 

Example transaction hash received on Rinkeby but not recorded on chain anywhere: `0x95b080cf5aea65f96ed0f01248f57f3b6657142db08a0072cad45be6b6ccf243`

**Comment by franckc** [comment]
@sparrowDom Sorry for I took so long to get back to you :( Yes, I was able to reproduce on iOS mobile app v0.24.3, using the truffle account and editing the listing 4-000-127.

**Comment by sparrowDom** [comment]
Fixed via: https://github.com/OriginProtocol/origin/pull/3725
